flask/launcher/build.gradle

```diff
@@ -0,0 +1,63 @@
+plugins {
+    id 'java-library'
+}
+
+evaluationDependsOn(":flask:common")
+
+sourceSets {
+    lockFileTest {
+    }
+}
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    ['', 'test', 'lockFileTest'].each {sourceSetName ->
+        ['compileOnly', 'annotationProcessor'].each { confBaseName ->
+            String configurationName = sourceSetName ? (sourceSetName + confBaseName.capitalize()) : confBaseName
+            add(configurationName, [group: "org.projectlombok", name: "lombok", version: lombok_version])
+        }
+    }
+    implementation group: "org.slf4j", name: "slf4j-api", version: slf4j_version
+    implementation group: "org.slf4j", name: "slf4j-simple", version: slf4j_version
+    implementation project(":flask:common")
+
+    testImplementation "org.junit.jupiter:junit-jupiter-api:$junit_jupiter_version"
+    testImplementation "org.junit.jupiter:junit-jupiter-params:$junit_jupiter_version"
+    testRuntimeOnly "org.junit.jupiter:junit-jupiter-engine:$junit_jupiter_version"
+
+    lockFileTestImplementation sourceSets.main.runtimeClasspath
+}
+
+tasks.register("tar", Tar) {
+    from(sourceSets.main.output)
+    from {
+        configurations['default'].collect { it.isDirectory() ? it : zipTree(it) }
+    }
+}
+
+Provider<Jar> lockFileTestExecutableJar = tasks.register("lockFileTestExecutableJar", Jar) {
+    manifest {
+        attributes "Main-Class": "net.corda.flask.launcher.LockFileTestMain"
+    }
+    from(sourceSets.lockFileTest.output)
+    from {
+        configurations['lockFileTestRuntimeClasspath'].collect {
+            if(it.exists()) {
+                it.isDirectory() ? it : zipTree(it)
+            } else {
+                null
+            }
+        }
+    }
+}
+
+test {
+    inputs.files(lockFileTestExecutableJar)
+    useJUnitPlatform()
+    systemProperty("lockFileTest.executable.jar", lockFileTestExecutableJar.get().outputs.files.singleFile.toString())
```

Comment:
I'll try to replace it with
```
systemProperty("lockFileTest.executable.jar", lockFileTestExecutableJar.map { it.outputs.files.singleFile.toString() })
```
praying that it still works

Reply:
I suspect not, because `map {}` will return another `Provider` which you would still need to `get()`. I suggested `doFirst {}`  because Gradle should execute it after the configuration phase is complete, which would make it safe to invoke `get()`.